File: src/main/java/com/github/zametki/model/GroupId.java
package com.github.zametki.model;

import com.github.mjdbc.DbMapper;
import com.github.mjdbc.Mapper;

public class GroupId extends AbstractId {

    public static final GroupId INVALID_ID = new GroupId(0);

    public GroupId(int value) {
        super(value);
    }

    @Mapper
    public static final DbMapper<GroupId> MAPPER = r -> new GroupId(r.getInt(1));
}

File: src/main/java/com/github/zametki/model/GroupId.kt
package com.github.zametki.model

import com.github.mjdbc.DbMapper
import com.github.mjdbc.Mapper

class GroupId(value: Int) : AbstractId(value) {
    companion object {

        val ROOT = GroupId(0)

        @Mapper
        @JvmField
        val MAPPER = DbMapper { r -> GroupId(r.getInt(1)) }
    }
}

